unstructured_ingest/v2/processes/connectors/vectara.py

```diff
@@ -0,0 +1,323 @@
+        """
+        Prepares dictionary in the format that Vectara requires
+
+        Select which meta-data fields to include and optionally map them to a new new.
+        remove the "metadata-" prefix from the keys
+        """
+        metadata_map = {
+            "page_number": "page_number",
+            "data_source-url": "url",
+            "filename": "filename",
+            "filetype": "filetype",
+            "last_modified": "last_modified",
+        }
+        md = flatten_dict(data, separator="-", flatten_lists=True)
+        md = {k.replace("metadata-", ""): v for k, v in md.items()}
+        md = {metadata_map[k]: v for k, v in md.items() if k in metadata_map}
+        return md
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        docs_list: Dict[Dict[str, Any]] = []
+        with open(elements_filepath) as elements_file:
+            elements_contents = json.load(elements_file)
+
+        docs_list: Dict[Dict[str, Any]] = []
+        conformed_elements = {
+            "documentId": str(uuid.uuid4()),
+            "title": elements_contents[0]
+            .get("metadata", {})
+            .get("data_source", {})
+            .get("record_locator", {})
+            .get("path"),
+            "section": [
+                {
+                    "text": element.pop("text", None),
+                    "metadataJson": json.dumps(self.conform_dict(data=element)),
+                }
+                for element in elements_contents
+            ],
+        }
+        logger.info(
+            f"Extending {len(conformed_elements)} json elements from content in {elements_filepath}"
+        )
+        docs_list.append(conformed_elements)
+
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        with open(output_path, "w") as output_file:
+            json.dump(docs_list, output_file)
+        return output_path
+
+
+class VectaraUploaderConfig(UploaderConfig):
+    batch_size: int = Field(default=50, description="Number of records per batch")
+
+
+@dataclass
+class VectaraUploader(Uploader):
+
+    connector_type: str = CONNECTOR_TYPE
+    upload_config: VectaraUploaderConfig
+    connection_config: VectaraConnectionConfig
+    _jwt_token: Optional[str] = field(init=False, default=None)
+    _jwt_token_expires_ts: Optional[float] = field(init=False, default=None)
+
+    def is_async(self) -> bool:
+        return True
+
+    def precheck(self) -> None:
+        try:
+            asyncio.run(self.vectara())
+        except Exception as e:
+            logger.error(f"Failed to validate connection {e}", exc_info=True)
+            raise DestinationConnectionError(f"failed to validate connection: {e}")
+
+    @property
+    async def jwt_token(self):
+        if not self._jwt_token or self._jwt_token_expires_ts - datetime.now().timestamp() <= 60:
```

Comment:
Does the jwt token have an expiration time? Otherwise we're hard coding 60 seconds here and that might be inaccurate.

Reply:
Maybe I'm misunderstanding, is this just checking that we're within a minute of it going to expire and refreshing it? If so, that's fine.